docs/limitations.md

```diff
@@ -0,0 +1,98 @@
+syscalls. Typically this can be worked around by spawning all processes directly
+from shadow's config file.  We plan to implement these calls:
+https://github.com/shadow/shadow/issues/1987.
+
+## IPv6
+
+Shadow does not yet implement IPv6. Most applications can be configured to use IPv4
+instead. Tracking issue: [#2216](https://github.com/shadow/shadow/issues/2216]).
+
+## Statically linked executables
+
+Shadow relies on `LD_PRELOAD` to inject code into the managed processes. This
+doesn't work for statically linked executables. Tracking issue:
+[#1839](https://github.com/shadow/shadow/issues/1839).
+
+Most applications can be dynamically linked, though occasionally you may need to
+edit build scripts and/or recompile.
+
+### golang
+
+`golang` typically defaults to producing statically linked executables, unless
+the application uses `cgo`. Using the networking functionality of `golang`'s
+standard library usually pulls in `cgo` by default and thus results in a
+dynamically linked executable.
+
+You can also explicitly force `go` to produce a dynamically linked executable. e.g.
+
+```
+# Install a dynamically linked `std`
+go install -buildmode=shared std
+# Build your application with dynamic linking
+go build -linkshared myapp.go
+```
+
+## Busy loops
+
+By default, Shadow runs each thread of managed processes until it's blocked by a
+syscall such as `nanosleep`,  `read`, `select`, `futex`, or `epoll`. Likewise,
+time only moves forward when Shadow is blocked on such a call - Shadow
+effectively models the CPU as being infinitely fast. This model is generally
+sufficient for modeling non-CPU-bound network applications.
+
+Unfortunately this model can lead to deadlock in the case of "busy loops", where
+a thread repeatedly checks for something to happen indefinitely or until some
+amount of wall-clock-time has passed. e.g., a worker thread might repeatedly
+check whether work is available for some amount of time before going to sleep on
+a `futex`, to avoid the latency of going to sleep and waking back up in cases
+where work arrives quickly. However since Shadow normally doesn't advance time
+when making non-blocking syscalls or allow other threads to run, such a loop can
+run indefinitely, deadlocking the whole simulation.
+
+When feasible, it's usually good practice to modify such loops to have a bound
+on the number of iterations instead of or in addition to a bound on wallclock
+time.
+
+For cases where modifying the loop is infeasible, Shadow provides the option
+`--model-unblocked-syscall-latency`. When this option is enabled, Shadow moves
+time forward a small amount on *every* syscall (and VDSO function call), and
+switches to another thread if one becomes runnable in the meantime (e.g. because
+network data arrived when the clock moved forward, unblocking it).
+
+This feature should only be used when it's needed to get around such loops. Some
+limitations:
```

Comment:
Is there a simulation performance impact?

Reply:
It can. I'll add a bullet for that.